core/src/impl/Kokkos_Core.cpp

```diff
@@ -282,11 +295,206 @@ void initialize_profiling(const InitArguments& args) {
+                                 "no");
+#endif
+#ifdef KOKKOS_ENABLE_PRAGMA_VECTOR
+  declare_configuration_metadata("vectorization", "KOKKOS_ENABLE_PRAGMA_VECTOR",
+                                 "yes");
+#else
+  declare_configuration_metadata("vectorization", "KOKKOS_ENABLE_PRAGMA_VECTOR",
+                                 "no");
+#endif
+
+#ifdef KOKKOS_ENABLE_HBWSPACE
+  declare_configuration_metadata("memory", "KOKKOS_ENABLE_HBWSPACE", "yes");
+#else
+  declare_configuration_metadata("memory", "KOKKOS_ENABLE_HBWSPACE", "no");
+#endif
+#ifdef KOKKOS_ENABLE_INTEL_MM_ALLOC
+  declare_configuration_metadata("memory", "KOKKOS_ENABLE_INTEL_MM_ALLOC",
+                                 "yes");
+#else
+  declare_configuration_metadata("memory", "KOKKOS_ENABLE_INTEL_MM_ALLOC",
+                                 "no");
+#endif
+#ifdef KOKKOS_ENABLE_POSIX_MEMALIGN
+  declare_configuration_metadata("memory", "KOKKOS_ENABLE_POSIX_MEMALIGN",
+                                 "yes");
+#else
+  declare_configuration_metadata("memory", "KOKKOS_ENABLE_POSIX_MEMALIGN",
+                                 "no");
+#endif
+
+#ifdef KOKKOS_ENABLE_ASM
+  declare_configuration_metadata("options", "KOKKOS_ENABLE_ASM", "yes");
+#else
+  declare_configuration_metadata("options", "KOKKOS_ENABLE_ASM", "no");
+#endif
+#ifdef KOKKOS_ENABLE_CXX14
+  declare_configuration_metadata("options", "KOKKOS_ENABLE_CXX14", "yes");
+#else
+  declare_configuration_metadata("options", "KOKKOS_ENABLE_CXX14", "no");
+#endif
+#ifdef KOKKOS_ENABLE_CXX17
+  declare_configuration_metadata("options", "KOKKOS_ENABLE_CXX17", "yes");
+#else
+  declare_configuration_metadata("options", "KOKKOS_ENABLE_CXX17", "no");
+#endif
+#ifdef KOKKOS_ENABLE_CXX20
+  declare_configuration_metadata("options", "KOKKOS_ENABLE_CXX20", "yes");
+#else
+  declare_configuration_metadata("options", "KOKKOS_ENABLE_CXX20", "no");
+#endif
+#ifdef KOKKOS_ENABLE_DEBUG_BOUNDS_CHECK
+  declare_configuration_metadata("options", "KOKKOS_ENABLE_DEBUG_BOUNDS_CHECK",
+                                 "yes");
+#else
+  declare_configuration_metadata("options", "KOKKOS_ENABLE_DEBUG_BOUNDS_CHECK",
+                                 "no");
+#endif
+#ifdef KOKKOS_ENABLE_HWLOC
+  declare_configuration_metadata("options", "KOKKOS_ENABLE_HWLOC", "yes");
+#else
+  declare_configuration_metadata("options", "KOKKOS_ENABLE_HWLOC", "no");
+#endif
+#ifdef KOKKOS_ENABLE_LIBRT
+  declare_configuration_metadata("options", "KOKKOS_ENABLE_LIBRT", "yes");
+#else
+  declare_configuration_metadata("options", "KOKKOS_ENABLE_LIBRT", "no");
+#endif
+#ifdef KOKKOS_ENABLE_MPI
+  declare_configuration_metadata("options", "KOKKOS_ENABLE_MPI", "yes");
+#else
+  declare_configuration_metadata("options", "KOKKOS_ENABLE_MPI", "no");
+#endif
+  declare_configuration_metadata("architecture", "Default Device",
+                                 typeid(Kokkos::DefaultExecutionSpace).name());
```

Comment:
This not a CMake option. We should likely remove or add it.

Reply:
This is being preserved from how the code used to work. It looks really dumb to me, but we used to print out this value in print_configuration. I'd prefer not to change this if we don't need to in this PR, we could do a follow-on

Reply:
Yes, absolutely. I didn't mean that you need to change anything here.